frontend/src/views/WheresMyGeneV2/components/GeneSearchBar/components/SourceDataButton/index.tsx

```diff
@@ -2,17 +2,30 @@ import { MouseEventHandler } from "react";
 import { track } from "src/common/analytics";
 import { EVENTS } from "src/common/analytics/events";
 import { EXCLUDE_IN_SCREENSHOT_CLASS_NAME } from "../SaveExport";
-import { StyledButtonDiv, StyledButtonIcon, StyledLabel } from "./style";
+import {
+  BadgeCounter,
+  StyledButtonDiv,
+  StyledButtonIcon,
+  StyledLabel,
+} from "./style";
 
 export default function SourceDataButton({
   handleRightSidebarButtonClick,
+  referenceCount,
```

Comment:
for simplicity, should we just call this `datasetsCount`? Or is there a reason we should use reference here instead? Thank you!